### DISCUSSION TITLE
Plans for iOS Text handling with external keyboards?

### DISCUSSION THREAD

**Discussion by Amenti** [AvaloniaUI/Avalonia#14951] [General]
I realized that some of the handling of external keyboard in TextBoxes in iOS is unfinished or broken. Backspace doesn't work correctly, and a lot of shortcuts don't work or lead to a crash.

I guess that's not really a bug but more like features which were not implemented yet. Is there any plan to alleviate these issues and fix the text handling for external keyboards in the near future?

I replaced all TextBoxes with native implementations for my app, but this leads to some airspace issues particularly with scrolling of the native TextBoxes outside of the visible area and I would love to switch back to the Avalonia TextBoxes.

**Comment by maxkatz6** [comment]
Please try 11.1-beta. Before that version there was zero handling of physical keyboards on ios.

**Reply by Amenti** [reply]
I forgot to mention that I was already using the Nightly Builds but did test out the Beta Nuget now. Backspace still doesn't work and some shortcuts e.g. Left + Shift crash the app.

**Reply by maxkatz6** [reply]
It might be very possible that some cases don't work yet, yes.
